### PR TITLE
Add compatibility section in README.md

### PR DESCRIPTION
**Pull request by pellared**
Structure and content based on the following:
- https://github.com/open-telemetry/opentelemetry-dotnet
- https://github.com/open-telemetry/opentelemetry-go

Initially, I have written that we support all Linux OSes that .NET supports. But together with @RassK we decided to be more hesitant and created https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/issues/282

### CODE REVIEW

**Review by dszmigielski — APPROVED** [review]

**Review by Kielek — APPROVED** [review]

File: docs/README.md [hunk 1]
```diff
@@ -21,8 +21,11 @@ except for versions lower than `.NET Framework 4.6.1`.
 
 ### Supported OSes
 
-OpenTelemetry .NET Auto-Instrumentation should work on all operating systems
-that given .NET version supports.
+The following operating systems are currently supported:
+
+- Windows,
+- MacOS,
+- Ubuntu.
```
Comment (pellared) [inline docs/README.md:29]:
```suggestion

However, it should also work on other Linux OSes that given .NET version supports.

```

@RassK Checked that it also works e.g. on Alpine. Do you think that adding ☝️  makes sense?

**Review by rajkumar-rangaraj — APPROVED** [review]

File: docs/README.md (hunk 1 above)
Comment (nrcventura) [reply docs/README.md:29]: I think that a statement like that is useful.

File: docs/README.md [hunk 2]
```diff
@@ -4,18 +4,35 @@
...
+### Supported .NET versions
+
+OpenTelemetry .NET Auto-Instrumentation attempts to work with all officially
+supported versions of [.NET Core](https://dotnet.microsoft.com/download/dotnet-core),
+and [.NET Framework](https://dotnet.microsoft.com/download/dotnet-framework)
+except for versions lower than `.NET Framework 4.6.1`.
+
+### Supported OSes
+
+The following operating systems are currently supported:
```
Comment (nrcventura) [inline docs/README.md:24]: When we use the word `supported` here, what are we implying? Are we implying that we have automated tests against these platforms, provide usage instructions and documentation, installation methods, or other things? Should we be more transparent about why we are calling out these operating systems?

Comment (pellared) [reply docs/README.md:24]: That is what I had in mind. Not sure if it is worth writing it down.

Comment (pjanotti) [reply docs/README.md:24]: @pellared it seems worth to spell it out. My suggestion would be to change the title to "Compatible OSes" and state that the project is automatically tested against such and such OSes and keep the line about that it should work with other Linux flavors.

Comment (pellared) [reply docs/README.md]: I tried to make it shorter and more concise

**Review by pjanotti — APPROVED** [review]

**Review by nrcventura — APPROVED** [review]

**Pull request merged**